app/controllers/companies_controller.rb

```diff
@@ -2,6 +2,48 @@
 
 class CompaniesController < ApplicationController
   def index
-    @companys = Company.all
+    @companies = Company.all
+  end
+
+  def show
+    @companies = Company.find(params[:id])
+  end
+  
+  def new
+    @companies = Company.new
+  end 
+
+  def create
+    @companies = Company.create(company_params)
+    if @companies.persisted?
+      redirect_to companies_path
+    end 
+  end
+
+  def edit
+    @companies = Company.find(params[:id])
+  end
+
+  def update
+    @companies= Company.find(params[:id])
+    if @companies.update(company_params)
+      redirect_to companies_path
+    else
+      render :edit, status: :unprocessable_entity
+    end 
+  end
+
+  def destroy
+    @companies= Company.find(params[:id])
+    @companies.destroy
+    redirect_to companies_path,
+    status: :see_other
+  end
+
+  private
+  def company_params
+    params.require(:company).permit(:name,:address)
   end
 end
+
+#def destroy @article = Article.find(params[:id]) @article.destroy redirect_to root_path, status: :see_other end private def article_params params.require(:article).permit(:title, :body) end
```

Comment:
just delete the commented code, the rest is fine.

Reply:
Done